### PR TITLE
[#5] Setup firebase distribution for CD

### LINKED ISSUE #5: [Chore] Setup Firebase project
## Why
To release builds to the testers

## Acceptance Criteria
- Setup Firebase project
- Application must be connected to the firebase console

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #5 

## What happened 👀

Application needs CD pipe line to distribute apk to the testers. Firebase App Distribution is the most popular and free for that purpose.

## Insight 📝

- Connect application with Firebase
- Add workflow for app distribution

## Proof Of Work 📹

TBD

### CODE REVIEW

File: .github/workflows/build_and_release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Build And Deploy App to Firebase Distribution
+
+on:
+  push:
+    branches:
+      - main
```
Comment (AVI5HEK) [inline .github/workflows/build_and_release.yml]: We usually have two separate deployment workflows for staging/dev and production. For this project, I think we should deploy the staging build every time a PR is merged to `develop`.

File: .github/workflows/build_and_release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Make gradlew executable
+        run: chmod +x ./gradlew
+      - name: build debug
+        run: ./gradlew assembleDevDebug
```
Comment (AVI5HEK) [inline .github/workflows/build_and_release.yml]: The release build should be the `release` build type. 😄 
```suggestion
        run: ./gradlew assembleDevRelease
```

File: app/google-services.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
```
Comment (AVI5HEK) [inline app/google-services.json]: This file should _NOT_ be added to VCS. It should be added to the Github Secrets (maybe as an encrypted base64 string), and then decoded in a step before building the app.

**Review by AVI5HEK — COMMENTED** [review]
Just some concerns for now.

File: app/google-services.json (hunk 3 above)
Comment (kaungkhantsoe) [reply app/google-services.json]: Updated @AVI5HEK

**Review by AVI5HEK — APPROVED** [review]
LGTM 🚀

File: .github/workflows/build_and_release.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: Build And Deploy App to Firebase Distribution
+
+on:
+  push:
+    branches:
+      - deveop
```
Comment (ryan-conway) [inline .github/workflows/build_and_release.yml]: I think this should say `develop` 😉

**Comment by kaungkhantsoe** [comment]
@ryan-conway Updated PR 🚀

**Review by ryan-conway — APPROVED** [review]

**Pull request merged**